numpyro/distributions/distribution.py

```diff
@@ -193,6 +193,16 @@ def event_dim(self):
         """
         return len(self.event_shape)
 
+    @property
+    def has_rsample(self):
+        return set(self.reparametrized_params) == set(self.arg_constraints)
```

Comment:
**[low priority]** I like @neerajprad's suggestion here. I suppose we could be even more precise; I don't know the JAX syntax, but in PyTorch we could write:
```py
return all(name in self.reparametrized_params
           for name in self.arg_constraints
           if getattr(self, name).requires_grad)
```

Reply:
It is trickier here because, in jax, the gradients are not stored in the variables: the function `grad(f)` can be applied for any ndarray.